packages/rest/lib/resource.cadl

```diff
@@ -110,3 +110,89 @@ interface ResourceInstanceOperations<TResource, TError> mixes ResourceRead<TReso
 interface ResourceCollectionOperations<TResource, TError> mixes ResourceCreate<TResource, TError>, ResourceList<TResource, TError> {}
 
 interface ResourceOperations<TResource, TError> mixes ResourceInstanceOperations<TResource, TError>, ResourceCollectionOperations<TResource, TError> {}
+
+interface SingletonResourceRead<TSingleton, TResource, TError> {
+  @autoRoute
+  @doc("Gets the singleton resource.")
+  @segmentOf(TSingleton)
+  @readsResource(TSingleton)
+  Get(...ResourceParameters<TResource>): TSingleton | TError;
+}
+
+interface SingletonResourceUpdate<TSingleton, TResource, TError> {
+  @autoRoute
+  @doc("Updates the singleton resource.")
+  @segmentOf(TSingleton)
+  @updatesResource(TSingleton)
+  Update(
+    ...ResourceParameters<TResource>,
+    @body properties: OptionalProperties<UpdateableProperties<TSingleton>>
+  ): TSingleton | TError;
+}
+
+interface SingletonResourceOperations<TSingleton, TResource, TError> mixes SingletonResourceRead<TSingleton, TResource, TError>, SingletonResourceUpdate<TSingleton, TResource, TError> {}
+
+interface ExtensionResourceRead<TExtension, TResource, TError> {
```

Comment:
The problem is that the ```TResource``` can often be any kind of resource - so that it doesn't have a fixed set of path parameters.  You usually *do not* want to model every possible target resource of an extension resource.  Instead, you want to say that there is a resource id (a multi-segment uri) that precedes the extension resource path parameters.

Reply:
That is certainly the case with ARM, but not what we've seen with the Language data plane service.  The aim of `ExtensionResource` in this case is to make it possible to add a standard set of fully-specified (i.e. the full URI path is present) extension operations on specific resource types.

For ARM, we'd have to represent this differently using the approach you're referring to.

Reply:
Gotcha, so if these extensions are only over known resources in the same data plane, that makes sense.